WPMedia/arc-themes-components
# Tmedia 660 prettier ticket

Testing Instructions: 

1. Add a comment in a file with lots of spaces 
2. Add and commit the file
3. See the pre-commit change https://github.com/WPMedia/arc-themes-components/pull/24/commits/8bf2c114dfe2413073746834422f22f00a497341
4. Locally, you should see linting (below) and prettier updates with vs code on save. I updated the vs code workspace settings
![Screen Shot 2022-02-22 at 09 18 04](https://user-images.githubusercontent.com/5950956/155165608-11bae35c-8ec7-46ba-bdb9-9daf73ce0a51.png)
5. `npm run format` should prettier-ify all compatible code 
6. `npm run lint` should have no contradictory issues with prettier due to eslint config 

Questions: 

- Double- or single-quotes? it looks like prettier has a default of double; this can be changed back 

-> Matthew prefers double 

- I can get lint-staged working on commit. I may need to do more research to get the actual formatting of code to work on push. It's super quick on commit though (and if you use vs code or ide formatting, you'll never see it). is this one on commit action ok? 

-> based on feedback, and the user experience, yes

AC: 

Install prettier 

Incorporate prettier with eslint via eslint-prettier

Formats all code in the repo 

On push, prettier should run and format the code and push that formatted code. (lint-staged does this well.) 

Applies to arc-themes-blocks, arc-themes-components, and engine-theme-sdk

Build works in engine-theme-sdk
